R/SS_doRetro.R

```diff
@@ -60,184 +53,76 @@
-
+                       RemoveBlocks = FALSE, verbose = FALSE, exe = "ss", ...) {
   # deprecated variable warnings -----
   # soft deprecated for now, but fully deprecate in the future.
-  if (lifecycle::is_present(intern)) {
+  if (lifecycle::is_present(masterdir)) {
     lifecycle::deprecate_warn(
-      when = "1.45.1",
-      what = "SS_doRetro(intern)",
-      details = "Please use show_in_console instead"
+      when = "1.46.0",
+      what = "SS_doRetro(masterdir)",
+      details = "Please use 'dir' instead"
     )
-    show_in_console <- !intern
+    dir <- masterdir
   }
 
-  # this should always be "windows" or "unix" (includes Mac and Linux)
-  OS <- .Platform[["OS.type"]]
-  # prefix applied to start of command to run model
-  prefix <- ifelse(OS == "windows", "", "./")
-
-  # save working directory
-  oldwd <- getwd()
-  on.exit(setwd(oldwd))
-
-  olddir <- file.path(masterdir, oldsubdir)
-  newdir <- file.path(masterdir, newsubdir)
-
-  # make directories, modify starter file, and start retrospective analyses
-
-  # for Windows users, automatically determine executable
-  if (!is.null(exefile) & OS == "windows") {
-    exefiles <- dir(olddir)[grep(".exe", dir(olddir))]
-    # if exactly one executable found in model directory, use that instead
-    if (length(exefiles) == 1) {
-      exefile <- exefiles
-    }
-    # for cases where there is more than one exe and the file name is
-    # "ss" by default, add the exe automatically
-    if (exefile == "ss") {
-      exefile <- "ss.exe"
-    }
-    if (!exefile %in% exefiles) {
-      stop("Missing executable file ", exefile, " in ", olddir)
-    }
-  }
+  olddir <- file.path(dir, oldsubdir)
+  newdir <- file.path(dir, newsubdir)
 
   # get model file names from olddir
   startfile <- dir(olddir)[tolower(dir(olddir)) == "starter.ss"]
-  forefile <- dir(olddir)[tolower(dir(olddir)) == "forecast.ss"]
-  wtatagefile <- dir(olddir)[tolower(dir(olddir)) == "wtatage.ss"]
-  testfile <- dir(olddir)[tolower(dir(olddir)) == "test.ss"]
-
-  if (length(startfile) == 0) stop("No starter.ss file found in ", olddir)
+  if (length(startfile) == 0) {
+    stop("No starter.ss file found in ", olddir)
+  }
 
+  # read original starter (later written to each folder)
   startfile <- file.path(olddir, startfile)
-
-  message("Getting input file names from starter file:\n", startfile)
   starter <- SS_readstarter(startfile, verbose = FALSE)
-  ctlfile <- starter[["ctlfile"]]
-  datfile <- starter[["datfile"]]
-
-  filenames <- c(exefile, forefile, ctlfile, datfile, wtatagefile, testfile)
-  message("copying model files from\n", olddir, "\n  to\n", newdir)
-  message("model files to copy:\n ", paste(filenames, collapse = "\n "))
-
-
-  if (!file.exists(newdir)) dir.create(newdir)
-
   subdirnames <- paste0(subdirstart, years)
 
+  # check for executable
+  check_exe(exe = exe, dir = olddir, verbose = verbose)
+
+  # loop over retrospective years
   for (iyr in 1:length(years)) {
```

Comment:
I suggest changing `1:length(years)` to `seq_along(years)`

Reply:
Good idea. I went ahead with a global find/replace for "1:length(" to "seq_along(" which found 132 instances.